app/controllers/profile/orders_controller.rb

```diff
@@ -31,13 +31,18 @@ def create
 
   def update
     order = Order.find(params[:id])
-    unless order.cancelled? || order.shipped?
+    role = current_user.role
+
+    if !order.cancelled? || !order.shipped?
```

Comment:
I think you can just check if the order is pending
```suggestion
    if order.pending?
```